subscriptions/providers/apple_in_app/__init__.py

```diff
@@ -127,12 +129,73 @@ def _get_plan_for_product_id(self, product_id: str) -> Plan:
         }
         return Plan.objects.get(**search_kwargs)
 
-    def _get_latest_transaction(self, original_transaction_id: str) -> SubscriptionPayment:
+    def _get_latest_transaction(self, original_transaction_id: str) -> Optional[SubscriptionPayment]:
+        # Latest transaction could be "CANCELLED", as it might be an upgrade.
+        # We assume that the user has a single subscription active for this app on the Apple platform.
```

Comment:
Can user have multiple subscriptions in the same app? If yes then you should either handle it correctly or, if it's complicated, put a big warning in the README.

Reply:
I've considered our options. Receipt is generated after the user has paid – so only a refund could be done (nothing was researched in that direction yet). Checking for multiple `original_transaction_id`s (it would mean that Apple allowed something nasty on their side) – even if we found them, only a refund could be done.

I've added a rather big warning statement in the readme, informing that the frontend app should try its best not to allow multiple subscriptions being bought by the same person. Even something as simple as "wait for response with subscription list before offering choice of subscriptions" would solve this issue.